src/main/java/com/ibm/cloud/sdk/core/util/CredentialUtils.java

```diff
@@ -214,6 +221,10 @@ protected static VcapService getVcapServiceEntry(String serviceName) {
       files.add(new File(userSpecifiedPath));
     }
 
+    if (StringUtils.isNotEmpty(userSpecifiedSystemProp)) {
```

Comment:
I couldn't move this to the end because unit tests would fail due to them finding other credential files in the java-core root directory

Reply:
yeah, I think this is fine.  If the user sets IBM_CREDENTIALS_FILE as an env var OR a system property, i think the intent is clear that they want to specify the custom filename as an override to any of the standard names we may look for.